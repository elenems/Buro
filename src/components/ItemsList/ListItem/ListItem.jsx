import React from "react";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import { makeStyles } from "@material-ui/core/styles";
import ScheduleIcon from "@material-ui/icons/Schedule";
import LocationOnOutlinedIcon from "@material-ui/icons/LocationOnOutlined";
import { Link } from "react-router-dom";

const useStyles = makeStyles({
  root: {
    maxWidth: "240px",
    padding: "8px 20px",
    margin: "20px 10px",
    boxShadow: "2px 2px 11px -4px grey"
  }
});

const makeFooterStyles = makeStyles({
  root: {
    marginTop: "6px",
    borderTop: "1px solid #03a9f4",
    paddingTop: "4px"
  }
});

export default function ListItem({ item }) {
  const classes = useStyles();
  const footerClasses = makeFooterStyles();
  return (
    <Box
      display="flex"
      flexDirection="column"
      justifyContent="space-evenly"
      classes={{ root: classes.root }}
    >
      <div className="imagePlaceholder">
        <Link to={`/item/${item.id}`}>
          <img src={item.imgUrl} alt={item.title} />
        </Link>
      </div>
      <Box classes={{ root: footerClasses.root }} p="0px 4px">
        <Typography variant="subtitle1" component="h4">
          {item.title}
        </Typography>
        <Box display="flex" alignItems="center" justifyContent="space-between">
          <Box display="flex" alignItems="center">
            <ScheduleIcon color="primary" />
            <Typography
              style={{ marginLeft: "4px" }}
              variant="subtitle1"
              component="span"
            >
              {item.date.slice(0,9)}
            </Typography>
          </Box>
          <Box display="flex" alignItems="center">
            <LocationOnOutlinedIcon color="primary" />
            <Typography variant="subtitle1" component="span">
              {item.location}
            </Typography>
          </Box>
        </Box>
      </Box>
    </Box>
  );
}
