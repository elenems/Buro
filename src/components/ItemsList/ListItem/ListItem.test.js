import React from "react";
import { shallow } from "enzyme";
import ListItem from "../ListItem";

describe("ListItem", () => {
  it("Matches snapshot", () => {
    const props = {
        item:{
            id:1,
            title:'Some title',
            date: new Date().toLocaleDateString()
        }
    }
    const wrapper = shallow(<ListItem {...props}/>).debug();
    expect(wrapper).toMatchSnapshot();
  });
});
