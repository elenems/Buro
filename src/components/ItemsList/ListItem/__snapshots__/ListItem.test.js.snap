// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Matches snapshot 1`] = `
"<Styled(MuiBox) display=\\"flex\\" flexDirection=\\"column\\" justifyContent=\\"space-evenly\\" classes={{...}}>
  <div className=\\"imagePlaceholder\\">
    <Link to=\\"/item/1\\">
      <img src={[undefined]} alt=\\"Some title\\" />
    </Link>
  </div>
  <Styled(MuiBox) classes={{...}} p=\\"0px 4px\\">
    <WithStyles(ForwardRef(Typography)) variant=\\"subtitle1\\" component=\\"h4\\">
      Some title
    </WithStyles(ForwardRef(Typography))>
    <Styled(MuiBox) display=\\"flex\\" alignItems=\\"center\\" justifyContent=\\"space-between\\">
      <Styled(MuiBox) display=\\"flex\\" alignItems=\\"center\\">
        <ScheduleIcon color=\\"primary\\" />
        <WithStyles(ForwardRef(Typography)) style={{...}} variant=\\"subtitle1\\" component=\\"span\\">
          2019-10-1
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
      <Styled(MuiBox) display=\\"flex\\" alignItems=\\"center\\">
        <LocationOnOutlinedIcon color=\\"primary\\" />
        <WithStyles(ForwardRef(Typography)) variant=\\"subtitle1\\" component=\\"span\\" />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Styled(MuiBox)>"
`;
